src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -604,24 +604,55 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
       } }
   }
 
-  private def setStatusAction(workflowId: WorkflowId, scopeKeys: Traversable[ExecutionDatabaseKey],
-                              callStatus: CallStatus)(implicit ec: ExecutionContext): DBIO[Unit] = {
-    // Describes a function from an input `Executions` to a projection of fields to be updated.
-    type ProjectionFunction = SlickDataAccess.this.dataAccess.Executions => (
-      Rep[String], Rep[Option[Timestamp]], Rep[Option[Int]], Rep[Option[String]], Rep[Option[String]], Rep[Option[Int]])
-    // If the call status is Starting, target the start date for update, otherwise target the end date.  The end date
-    // is only set to a non-None value if the status is terminal.
-    val projectionFn: ProjectionFunction = if (callStatus.isStarting)
-      e => (e.status, e.startDt, e.rc, e.executionHash, e.dockerImageHash, e.resultsClonedFrom)
-    else
-      e => (e.status, e.endDt, e.rc, e.executionHash, e.dockerImageHash, e.resultsClonedFrom)
-
-    val maybeDate = if (callStatus.isStarting || callStatus.isTerminal) Option(new Date().toTimestamp) else None
-
-    // If this call represents a call caching hit, find the execution ID for the call from which results were cloned and
-    // wrap that in an `Option`.
-    // If this wasn't a call caching hit just return `DBIO.successful(None)`, `None` lifted into `DBIO`.
-    val findResultsClonedFromId = callStatus.resultsClonedFrom map { backendCall =>
+  private def executionsForStatusUpdate(workflowId: WorkflowId, scopeKeys: Traversable[ExecutionDatabaseKey])(implicit ec: ExecutionContext) = for {
+    workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
+    executions = dataAccess.executionsByWorkflowExecutionIdAndScopeKeys(workflowExecutionResult.workflowExecutionId.get, scopeKeys)
+  } yield executions
+
+  private def assertUpdateCount(updates: Int, expected: Int) = require(updates == expected, s"Execution update count $updates did not match scopes size $expected")
+
+  def setStartingStatus(workflowId: WorkflowId, scopeKeys: Traversable[ExecutionDatabaseKey])(implicit ec: ExecutionContext): Future[Unit] = {
+    if (scopeKeys.isEmpty) Future.successful(()) else runTransaction(setStartingStatusAction(workflowId, scopeKeys))
+  }
+
+  def setStartingStatusAction(workflowId: WorkflowId, scopeKeys: Traversable[ExecutionDatabaseKey])(implicit ec: ExecutionContext) = {
+    for {
+      executions <- executionsForStatusUpdate(workflowId, scopeKeys)
+      count <- executions.map(e => (e.status, e.startDt)).update((ExecutionStatus.Starting.toString, Option(new Date().toTimestamp)))
+      _ = assertUpdateCount(count, scopeKeys.size)
+    } yield ()
+  }
+
+  def setOtherStatus(workflowId: WorkflowId, scopeKeys: Traversable[ExecutionDatabaseKey], status: ExecutionStatus)(implicit ec: ExecutionContext): Future[Unit] = {
```

Comment:
I think I prefer "updateStatus" - since we can always assume this is an update of an existing value, I think?